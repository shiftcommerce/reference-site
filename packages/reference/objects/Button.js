// Libraries
import { Component } from 'react'
import classNames from 'classnames'

class Button extends Component {
  render () {
    let {
      status,
      label,
      size,
      disabled,
      className,
      ...otherProps
    } = this.props

    disabled = disabled || false
    status = status || 'primary'
    size = size || 'lrg'

    return (
      <button
        className={classNames(className, 'o-button', `o-button--${status}`, `o-button--${size}`, {'o-button--disabled': disabled})}
        tabIndex='0'
        role='button'
        disabled={disabled}
        {...otherProps}
      >
        { label }
      </button>
    )
  }
}

export default Button
