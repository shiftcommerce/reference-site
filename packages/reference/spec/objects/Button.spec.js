// Object
import Button from '../../objects/Button'

test('renders correctly', () => {
  // Arrange & Act
  const wrapper = mount(
    <Button label='ADD TO BAG' status='positive' />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
})
