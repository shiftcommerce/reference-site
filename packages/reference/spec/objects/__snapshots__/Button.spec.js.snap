// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Button
  label="ADD TO BAG"
  status="positive"
>
  <button
    className="o-button o-button--positive"
    role="button"
    tabIndex="0"
  >
    ADD TO BAG
  </button>
</Button>
`;
